internal/pkg/comm/connection.go

```diff
@@ -75,10 +75,18 @@ func (cs *CredentialSupport) GetPeerCredentials() credentials.TransportCredentia
 	})
 }
 
-func (cs *CredentialSupport) AppRootCAsByChain() map[string][][]byte {
+func (cs *CredentialSupport) AppRootCAsByChain() [][]byte {
 	cs.mutex.RLock()
 	defer cs.mutex.RUnlock()
-	return cs.appRootCAsByChain
+	var trustedRoots [][]byte
+	// iterate over all roots for all app and orderer channels
+	for _, roots := range cs.appRootCAsByChain {
+		trustedRoots = append(trustedRoots, roots...)
+	}
+
+	trustedRootsCopy := make([][]byte, len(trustedRoots))
+	copy(trustedRootsCopy, trustedRoots)
```

Comment:
```suggestion
	trustedRoots := make([][]byte, 0, len(cs.appRootCAsByChain))
	for _, roots := range cs.appRootCAsByChain {
		trustedRoots = append(trustedRoots, roots...)
	}

	return append([][]byte(nil), trustedRoots...)
}```

I think you can do it simpler a bit.

Reply:
I think you can make it a little bit simpler.

Reply:
or even 

```
	trustedRoots := make([][]byte, 0, len(cs.appRootCAsByChain))
	for _, roots := range cs.appRootCAsByChain {
		trustedRoots = append(trustedRoots, roots...)
	}

	return trustedRoots
```

Reply:
Fixed to the second option